File: linked_list/java/ListNode.java

public class ListNode
{
	public int val;
	public ListNode next;
	public ListNode(int x) { val = x; next = null; }
	public static ListNode getLinkedList(int[] array)
	{
		if(array.length == 0)
			return null;
		ListNode head = new ListNode(array[0]);
		ListNode curr = head;
		for(int i = 1; i < array.length; i++)
		{
			curr.next = new ListNode(array[i]);
			curr = curr.next;
		}
		return head;
	}
	public static int[] getArray(ListNode head)
	{
		ListNode curr = head;
		int cnt = 0;
		while(curr != null)
		{
			curr = curr.next;
			cnt++;
		}
		int[] array = new int[cnt];
		for(int i = 0; i < cnt; i++)
		{
			array[i] = curr.val;
			curr = curr.next;
		}
		return array;
	}
}

File: sorting/py/ListNode.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

def getLinkedList(array):
    head = ListNode(array[0])
    ptr = head
    for i in range(1, len(array)):
        ptr.next = ListNode(array[i])
        ptr = ptr.next
    return head

def getArray(head):
    ptr = head
    arr = []
    while ptr:
        try:
            arr.append(ptr.val)
        except:
            print ptr
        ptr = ptr.next
    return arr
